### PR TITLE
[4.0][api][com_content] some fixes

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes

1. added `fulltext` field to the articles model select list
2. fix warning `array_combine()`
3.fix PHP Notice When providing non existent article ID

### Testing Instructions

1. call the api `api/index.php/v1/content/article`
2. call the api `api/index.php/v1/content/article/XX`

where xx is an article id without tags
and/or
where xx is not an article existent article id

### Expected result

no notice
no warning

### Actual result
`PHP Notice:  Undefined property: stdClass::$fulltext in .../api/components/com_content/View/Articles/JsonapiView.php on line 138
`

`PHP Warning:  array_combine(): Both parameters should have an equal number of elements in .../api/components/com_content/View/Articles/JsonapiView.php on line 171`

### CODE REVIEW

File: api/components/com_content/View/Articles/JsonapiView.php [hunk 1]
```diff
@@ -163,7 +163,7 @@ protected function prepareItem($item)
 			$item->associations = $associations;
 		}
 
-		if (!empty($item->tags))
+		if ((!empty($item->tags)) && (!empty($item->tags->tags)))
```
Comment (Quy) [inline api/components/com_content/View/Articles/JsonapiView.php]:
```suggestion
		if (!empty($item->tags->tags))
```
Will this work?

Comment (alikon) [reply api/components/com_content/View/Articles/JsonapiView.php]: yes, but it is safe ?

Comment (Quy) [reply api/components/com_content/View/Articles/JsonapiView.php]: It should. `empty` checks if it exists and if empty.

Comment (alikon) [reply api/components/com_content/View/Articles/JsonapiView.php]: ok then

**Comment by Quy** [comment]
When providing nonexistent article ID: 

> PHP Notice:  Undefined property: stdClass::$rawvalue in \api\components\com_content\View\Articles\JsonapiView.php on line 146

File: api/components/com_content/View/Articles/JsonapiView.php [hunk 2]
```diff
@@ -135,6 +135,11 @@ public function displayItem($item = null)
 	 */
 	protected function prepareItem($item)
 	{
+		if ($item->id === null)
+		{
+			return $item;
+		}
+
```
Comment (Quy) [inline api/components/com_content/View/Articles/JsonapiView.php]: Remove tabs.

Comment (alikon) [reply api/components/com_content/View/Articles/JsonapiView.php]: done

**Comment by Quy** [comment]
Sorry more on multilingual site:

> PHP Notice:  Undefined property: stdClass::$associations in \api\components\com_content\View\Articles\JsonapiView.php on line 158

> PHP Warning:  Invalid argument supplied for foreach() in \api\components\com_content\View\Articles\JsonapiView.php on line 158

**Comment by alikon** [comment]
no problem at all
happy that you are  testing...
just keep patience with my dirty hacks :smile:

**Comment by alikon** [comment]
just give me more hints on how to replicate.... i'm on a multiligual now, but still don't get that notice with this pr applyed..... what i'm missing ?

**Comment by Quy** [comment]
Install French and German.
Install Multilingual Sample Data.
Call the API.

**Comment by alikon** [comment]
can you re-test please

File: api/components/com_content/View/Articles/JsonapiView.php [hunk 3]
```diff
@@ -146,10 +151,10 @@ protected function prepareItem($item)
 			$item->{$field->name} = isset($field->apivalue) ? $field->apivalue : $field->rawvalue;
 		}
 
-		if (\Joomla\CMS\Language\Multilanguage::isEnabled())
-		{
-			$associations = [];
+		$associations = [];
```
Comment (Quy) [inline api/components/com_content/View/Articles/JsonapiView.php]: How about moving lines 154 and 169 inside the if statement?

Comment (alikon) [reply api/components/com_content/View/Articles/JsonapiView.php]: moved

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 5f1cd12bc5495b59bec9739cd937e19b1c5cdb12<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27071">issues.joomla.org/tracker/joomla-cms/27071</a>.</sub>

**Comment by alikon** [comment]
as issues are still present 
better to do another fresh one to solve conflict

**Pull request closed without merging**